src/runner/bootstrapper.js

```diff
@@ -82,7 +82,7 @@ export default class Bootstrapper {
             tests = testsWithOnlyFlag;
 
         if (this.filter)
-            tests = tests.filter(test => this.filter(test.name, test.fixture.name, test.fixture.path));
+            tests = tests.filter(test => this.filter(test.name, test.fixture.name, test.fixture.path, Object.assign({}, test.fixture.meta, test.meta)));
```

Comment:
We've discussed your PR with the team and decided to provide two separate options:

`--testMeta`
`--fixtureMeta`

Here's why.

1) Right now, we don't mix fixture and test metadata in reports ([link1](https://github.com/DevExpress/testcafe/commit/d4be3b190755b613e73ae3c69894ef523b55ff0d#diff-72061a2448a892e4baf1d16632a044adR58), [link2](https://github.com/DevExpress/testcafe/commit/d4be3b190755b613e73ae3c69894ef523b55ff0d#diff-72061a2448a892e4baf1d16632a044adR77)).
2) It's more clear which meta (test or fixture) is used for filtering tests to run.
3) To prevent collisions when fixture and test have the same metadata record.
 
The final API will look as follows:

Programmatic:

`.filter((testName, fixtureName, fixturePath, testMeta, fixtureMeta) => {})`
 
CLI:

`testcafe chrome tests/ --testMeta "{ "prod": "true"}" --fixtureMeta "{"mobile": "false"}"`

Reply:
It seems a good idea!.

I'll use snake case for the cli parameters, in order to be coherent with the naming of the other parameters:
- `--test-meta`
- `--fixture-meta`